src/data/postgres.py

```diff
@@ -174,107 +176,6 @@ def close_session(session: Session) -> None:
-        session.rollback()
-        print(f"{wallet_infos=}")
-        raise err
-
-
-def add_pool_config(pool_config: PoolConfig, session: Session) -> None:
-    """Add pool config to the pool config table if not exist.
-
-    Verify pool config if it does exist.
-
-    Arguments
-    ---------
-    pool_config : PoolConfig
-        A PoolConfig object to insert into postgres
-    session : Session
-        The initialized session object
-    """
-    # NOTE the logic below is not thread safe, i.e., a race condition can exists
-    # if multiple threads try to add pool config at the same time
-    # This function is being called by acquire_data.py, which should only have one
-    # instance per db, so no need to worry about it here
-
-    # Since we're doing a direct equality comparison, we don't want to coerce into floats here
-    existing_pool_config = get_pool_config(session, contract_address=pool_config.contractAddress, coerce_float=False)
-
-    if len(existing_pool_config) == 0:
-        session.add(pool_config)
-        try:
-            session.commit()
-        except exc.DataError as err:
-            session.rollback()
-            print(f"{pool_config=}")
-            raise err
-    elif len(existing_pool_config) == 1:
-        # Verify pool config
-        for key in PoolConfig.__annotations__.keys():
-            new_value = getattr(pool_config, key)
-            old_value = existing_pool_config.loc[0, key]
-            if new_value != old_value:
-                raise ValueError(
-                    f"Adding pool configuration field: key {key} doesn't match (new: {new_value}, old:{old_value})"
-                )
-    else:
-        # Should never get here, contractAddress is primary_key, which is unique
-        raise ValueError
-
-
-def add_pool_infos(pool_infos: list[PoolInfo], session: Session) -> None:
-    """Add a pool info to the poolinfo table.
-
-    Arguments
-    ---------
-    pool_infos : list[PoolInfo]
-        A list of PoolInfo objects to insert into postgres
-    session : Session
-        The initialized session object
-    """
-    for pool_info in pool_infos:
-        session.add(pool_info)
-    try:
-        session.commit()
-    except exc.DataError as err:
-        session.rollback()
-        print(f"{pool_infos=}")
-        raise err
-
-
-def add_checkpoint_infos(checkpoint_infos: list[CheckpointInfo], session: Session) -> None:
-    """Add checkpoint info to the checkpointinfo table.
-
-    Arguments
-    ---------
-    checkpoint_infos : list[CheckpointInfo]
-        A list of CheckpointInfo objects to insert into postgres
-    session : Session
-        The initialized session object
-    """
-    for checkpoint_info in checkpoint_infos:
-        session.add(checkpoint_info)
-    try:
-        session.commit()
-    except exc.DataError as err:
-        session.rollback()
-        raise err
-
-
 def add_transactions(transactions: list[Transaction], session: Session) -> None:
```

Comment:
I'd argue that these transactions isn't at all generic, but instead specific to hyperdrive. This is due to decoding of logs. Fields of `Transaction` include specific columns that match the input parameters to methods (e.g., openLong)

Reply:
yes very true, I didn't scroll down enough.